internal/controller/resourceset_controller_test.go

```diff
@@ -641,6 +641,266 @@ spec:
 	g.Expect(conditions.GetMessage(result, meta.ReadyCondition)).To(ContainSubstring("failed to parse expression"))
 }
 
+func TestResourceSetInputsFromValidation(t *testing.T) {
+	g := NewWithT(t)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	ns, err := testEnv.CreateNamespace(ctx, "test")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Both set.
+	err = testEnv.Create(ctx, &fluxcdv1.ResourceSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: ns.Name,
+		},
+		Spec: fluxcdv1.ResourceSetSpec{
+			InputsFrom: []fluxcdv1.InputProviderReference{{
+				Kind:     fluxcdv1.ResourceSetInputProviderKind,
+				Name:     "test",
+				Selector: &metav1.LabelSelector{},
+			}},
+		},
+	})
+	g.Expect(err).To(HaveOccurred())
+	g.Expect(err.Error()).To(ContainSubstring("cannot set both name and selector for input provider references"))
+
+	// Neither set.
+	err = testEnv.Create(ctx, &fluxcdv1.ResourceSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: ns.Name,
+		},
+		Spec: fluxcdv1.ResourceSetSpec{
+			InputsFrom: []fluxcdv1.InputProviderReference{{
+				Kind: fluxcdv1.ResourceSetInputProviderKind,
+			}},
+		},
+	})
+	g.Expect(err).To(HaveOccurred())
+	g.Expect(err.Error()).To(ContainSubstring("at least one of name or selector must be set for input provider references"))
+}
+
+func TestResourceSetReconciler_LabelSelector(t *testing.T) {
```

Comment:
Please add a new test with for GC with just label selectors:
- Start with no RSIPs - RSET should reconcile with an empty inventory
- Add a couple of RSIPs that match the selector - check RSET inventory is filled 
- Delete one RSIP - check RSET inventory was updated
- Delete all RSIPs - RSET should reconcile with an empty inventory and GC should delete all items